components/event-feed-service/pkg/persistence/project_update.go

```diff
@@ -0,0 +1,215 @@
+			Priority: 1,
+		},
+	}, nil
+}
+
+func (efs ElasticFeedStore) RunProjectUpdateTask(ctx context.Context, projectUpdateID string, params map[string]string,
+	projectTaggingRules map[string]*authz.ProjectRules) (project_update_lib.SerializedProjectUpdateTaskID, project_update_lib.SerializedProjectUpdateTaskStatus, error) {
+	logrus.Debug("running RunProjectUpdateTask")
+
+	if params == nil {
+		return "", project_update_lib.NoStatus, errors.New("project update tasks params missing")
+	}
+
+	indexName := params["n"]
+	if indexName == "" {
+		return "", project_update_lib.NoStatus, errors.New("project update task index missing")
+	}
+
+	if indexName == IndexNameFeeds {
+		id, err := efs.updateEventProjectTags(ctx, projectTaggingRules)
+		if err != nil {
+			return "", project_update_lib.NoStatus, err
+		}
+
+		return project_update_lib.SerializedProjectUpdateTaskID(id),
+			project_update_lib.SerializedProjectUpdateTaskStatus{
+				State: project_update_lib.SerializedProjectUpdateTaskRunning,
+			}, nil
+	}
+
+	return "", project_update_lib.NoStatus, errors.Errorf(
+		"project update index %s does not match any current mappings", indexName)
+}
+
+func (efs ElasticFeedStore) MonitorProjectUpdateTask(ctx context.Context, projectUpdateID string,
+	id project_update_lib.SerializedProjectUpdateTaskID) (project_update_lib.SerializedProjectUpdateTaskStatus, error) {
+	logrus.Debug("running MonitorProjectUpdateTask")
+
+	res, err := elasticutil.GetUpdateByQueryTask(ctx, efs.client, string(id))
+
+	if err != nil {
+		logrus.WithError(err).Error("Could not query task")
+		return project_update_lib.NoStatus, err
+	}
+
+	projectUpdateStatus := project_update_lib.SerializedProjectUpdateTaskStatus{}
+	if res.Completed {
+		projectUpdateStatus.PercentageComplete = 1
+		if res.Response == nil {
+			projectUpdateStatus.State = project_update_lib.SerializedProjectUpdateTaskSuccess
+			logrus.Warnf("response missing from completed task for task %s", id)
+		} else if len(res.Response.Failures) > 0 {
+			projectUpdateStatus.State = project_update_lib.SerializedProjectUpdateTaskFailed
+			projectUpdateStatus.Error = res.Response.Failures[0]
+		} else {
+			projectUpdateStatus.State = project_update_lib.SerializedProjectUpdateTaskSuccess
+		}
+	} else {
+		projectUpdateStatus.State = project_update_lib.SerializedProjectUpdateTaskRunning
+		if res.Task.Status != nil && res.Task.Status.Total > 0 {
+			projectUpdateStatus.PercentageComplete = float32(
+				res.Task.Status.Updated+res.Task.Status.Created+res.Task.Status.Deleted+
+					res.Task.Status.VersionConflicts+res.Task.Status.Noops) / float32(res.Task.Status.Total)
+		}
+	}
+
+	return projectUpdateStatus, nil
+}
+
+func (efs ElasticFeedStore) CancelProjectUpdateTask(ctx context.Context,
+	projectUpdateID string, id project_update_lib.SerializedProjectUpdateTaskID) error {
+	logrus.Debug("running CancelProjectUpdateTask")
+	return efs.JobCancel(ctx, string(id))
+}
+
+func (efs ElasticFeedStore) updateEventProjectTags(
```

Comment:
This function does the project update. The code in the script variable is sent to elasticsearch and run on each event.